modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaSearchVocabulariesMethodCheck.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaSearchVocabulariesMethodCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		String newContent = content;
+
+		Matcher matcher = _searchVocabulariesPattern.matcher(content);
+
+		while (matcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, matcher.start());
+
+			if (!hasClassOrVariableName(
+					"AssetVocabularyService", content, methodCall) &&
+				!hasClassOrVariableName(
+					"AssetVocabularyLocalService", content, methodCall)) {
```

Comment:
@TamyrisTorres, is there any special reason for you apply different checks for similar use cases? Please see the image below:

![image](https://github.com/natocesarrego/liferay-portal/assets/6729287/50e2befc-46fc-4136-ac52-59f2c608b9a2)

Can you please give me more details?

Thanks

Reply:
@natocesarrego 

This case is different because when I have the declaration of the variable in the class I use the hasClassOrVariableName method. Already when my test uses the call the class for calling the method I Can´t use the hasClassOrVariableName, I need to use the getVariableName method.